### DISCUSSION TITLE
dfu-util user error?

### DISCUSSION THREAD

**Discussion by squaregrooper** [Ralim/IronOS#1269] [Q&A] [answered]
Im getting the below error when I try to flash, ive tried updating LIBUSB to the dev branch and still have no luck, anyone have any hints?

Im using the latest version of dfu-util.  The firmware has been flashed once before, but it was a while ago.  Im trying to get the build/flash steps down before I modify things.

`me@me:$ dfu-util -D ~/IronOS/source/Hexfile/Pinecil_EN.dfu
dfu-util 0.11

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2021 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Match vendor ID from file: 28e9
Match product ID from file: 0189
dfu-util: Cannot open DFU device 28e9:0189 found on devnum 8 (LIBUSB_ERROR_ACCESS)
dfu-util: No DFU capable USB device available`

**Comment by Ralim** [comment, marked as answer]
`LIBUSB_ERROR_ACCESS`
means you dont have permissions; try with `sudo`

**Reply by squaregrooper** [reply]
thanks, i could have sworn i tried this but it worked.  I noticed i had to run the program pretty immediately after plugging the iron in, if I waited more than a few seconds it would error out with almost the same error, this time the `LIBUSB_ERROR_IO` error

`me@me:~$ sudo dfu-util -D ~/IronOS/source/Hexfile/Pinecil_EN.dfu
dfu-util 0.11

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2021 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Match vendor ID from file: 28e9
Match product ID from file: 0189
dfu-util: Cannot open DFU device 28e9:0189 found on devnum 22 (LIBUSB_ERROR_IO)
dfu-util: No DFU capable USB device available`
